grails-app/services/com/unifina/service/StreamService.groovy

```diff
@@ -128,40 +120,41 @@ class StreamService {
 
 	// Ref to Kafka will be abstracted out when Feed abstraction is reworked
 
-	void sendMessage(Stream stream, String partitionKey, long timestamp, byte[] content, byte contentType, int ttl=0,
-					 SignatureType signatureType = SignatureType.SIGNATURE_TYPE_NONE, String address = null, String signature = null) {
+	void sendMessage(Stream stream, String partitionKey, long timestamp, String content, StreamMessage.ContentType contentType,
```

Comment:
Since the sender is now stateful (must maintain `sequenceNumber` and `previousMsgRef`), meaning the sender has a lot of responsibility for constructing a correct message anyway, wouldn't it be nice to just have `sendMessage(StreamMessage)`? Could that be pulled off?

Now you're hard-coding the sequenceNumber to 0 and previousMsgRef to null, and defaulting address to `""`, but would be good to get proper values in them. Passing the responsibility to whoever calls this and exposing just `sendMessage(StreamMessage)` would enable/encourage/force that.